onmt/inputters/text_utils.py

```diff
@@ -16,15 +41,22 @@ def text_sort_key(ex):
 def clean_example(maybe_example):
     maybe_example['src'] = {"src": ' '.join(maybe_example['src'])}
     # Make features part of src like
-    # {'src': {'src': ..., 'feat1': ...., 'feat2': ....}}
-    if 'src_feats' in maybe_example:
-        for feat_name, feat_value in maybe_example['src_feats'].items():
-            maybe_example['src'][feat_name] = ' '.join(feat_value)
-        del maybe_example['src_feats']
+    # {'src': {'src': ..., 'feats': ....}
+    maybe_example['src']['feats'] = \
+        [' '.join(x) for x in maybe_example["src_feats"]]
```

Comment:
are we 100% sure the key 'feats' always exists ?

Reply:
`maybe_example["src_feats"]` will be an empty list in those cases, and so will be `maybe_example['src']['feats']`. Maybe it's better to remove it from the dict?